### PR TITLE
GAP-2392: Build app dashboard UI changes

### PR DESCRIPTION
**Pull request by dominicwest**
## Description

Ticket [GAP-2392](https://technologyprogramme.atlassian.net/jira/software/projects/GAP/boards/216?selectedIssue=GAP-2392)

Summary of the changes and the related issue. List any dependencies that are required for this change:
- UI changes to build app dashboard page
- New edit section page
- Amended flows to navigate to this page appropriately

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [x] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

Screenshot 2024-01-29 at 12 08 04
Screenshot 2024-01-29 at 12 08 01
Screenshot 2024-01-29 at 12 07 52

https://github.com/cabinetoffice/gap-find-apply-web/assets/101722961/bad0d55e-ec5f-4c68-be21-5fe3ffe48acb

# Checklist:

- [x] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [x] I have performed a self-review of my code.
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-2392]: https://technologyprogramme.atlassian.net/browse/GAP-2392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    });
+
+    it('Should render a button to save this section', () => {
+      expect(
+        screen.getByRole('button', { name: 'Save and go back' })
+      ).toHaveAttribute(
+        'href',
+        '/apply/build-application/some-application-id/dashboard'
+      );
+    });
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx:203]:
```suggestion
it.each([ 
  ["Add a new question", “/apply/build-application/some-application-id/testSectionId/question-content”], 
  ["Delete section", “/apply/build-application/some-application-id/testSectionId/delete-confirmation”],
  ["Save and go back", “/apply/build-application/some-application-id/dashboard”]
])(“button %s should have href %s”, 
 (name, href) => { 
    expect(screen.getByRole('button', { name })).toHaveAttribute('href', href);
  }
);
```

File: packages/gap-web-ui/src/components/table/Table.module.scss [hunk 2]
```diff
@@ -16,3 +16,7 @@
 .gap-table-cell-wrap-text {
   word-break: break-word;
 }
+
+.disable-bottom-row-border {
+  border-bottom: none !important;
```
Comment (ConorFayleTCO) [inline packages/gap-web-ui/src/components/table/Table.module.scss:21]: is there any way we can do this without !important?

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx (hunk 1 above)
Comment (dominicwest) [reply packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx:203]: Neat - didnt know you could do this <3 formatter kinda rips it to shreds, but still worth it

File: packages/gap-web-ui/src/components/table/Table.module.scss (hunk 2 above)
Comment (dominicwest) [reply packages/gap-web-ui/src/components/table/Table.module.scss:21]: I'll play about with it again but I don't believe so

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx (hunk 1 above)
Comment (ConorFayleTCO) [reply packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx]: yeah test.each is great for this kind of thing and doesn't seem widely known
https://elfi-y.medium.com/reduce-boilerplate-test-code-with-jest-it-each-30a0eec9776d

**Review by ConorFayleTCO — CHANGES_REQUESTED** [review]
Looks like there's some styling issues when compared to the figma - the questions need a bit more room to breathe in their rows and should be aligned with the card header, the question title should be bold, and there's some issues with vertical spacing between elements at the bottom of the page
https://www.figma.com/file/e0eCBeGIKxM6rzWAR23CGN/Admin?type=design&node-id=108-9585&mode=design&t=ZNzHQoUsN9BJ9Voj-0

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/delete-confirmation.page.tsx [hunk 3]
```diff
@@ -57,7 +58,10 @@ export const getServerSideProps: GetServerSideProps = async ({
         });
       }
     },
-    `/build-application/${applicationId}/dashboard`,
+    (deleted) =>
+      deleted
+        ? `/build-application/${applicationId}/dashboard`
+        : `/build-application/${applicationId}/${sectionId}`,
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/delete-confirmation.page.tsx:64]: can we put these in routes?

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/delete-confirmation.page.tsx [hunk 4]
```diff
@@ -70,7 +74,7 @@ export const getServerSideProps: GetServerSideProps = async ({
   return {
     props: {
       fieldErrors,
-      backButtonHref: `/build-application/${applicationId}/dashboard`,
+      backButtonHref: `/build-application/${applicationId}/${sectionId}`,
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/delete-confirmation.page.tsx:77]: again, routes if possible

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+  section,
+  grantApplicationName,
+  applicationId,
+}: InferProps<typeof getServerSideProps>) => {
+  return (
+    <>
+      <Meta title="Build an application form - Manage a grant" />
+
+      <CustomLink
+        href={`/build-application/${applicationId}/dashboard`}
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx:54]: routes

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            isVisuallyHidden: true,
+            width: 'one-third',
+          },
+          { name: 'Actions', isVisuallyHidden: true, width: 'one-third' },
+        ]}
+        rows={questionTableRows(section)}
+      />
+
+      <CustomLink
+        href={`/build-application/${applicationId}/${section.sectionId}/question-content`}
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx:107]: ^

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+      <h2 className="govuk-heading-m">Delete this section</h2>
+
+      <p className="govuk-caption-m">
+        Any questions or information in this section will be deleted
+        permanently.
+      </p>
+
+      <CustomLink
+        href={`/build-application/${applicationId}/${section.sectionId}/delete-confirmation`}
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx:123]: ^

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+        isButton
+        customStyle="govuk-button--warning"
+      >
+        Delete section
+      </CustomLink>
+
+      <hr className="govuk-section-break govuk-section-break--m govuk-section-break--visible" />
+
+      <CustomLink
+        href={`/build-application/${applicationId}/dashboard`}
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx:133]: ^

File: packages/admin/src/pages/build-application/[applicationId]/components/Sections.tsx [hunk 9]
```diff
@@ -20,151 +26,65 @@ const Sections = ({
...
+                        <div className="govuk-grid-column-one-half">
+                          {sectionIndex + 1}. {section.sectionTitle}
+                        </div>
+                        <div className="govuk-grid-column-one-half">
+                          <p className="govuk-!-text-align-right govuk-!-font-size-19 govuk-!-margin-0">
                             <CustomLink
-                              href={`/build-application/${applicationId}/${section.sectionId}/${question.questionId}/delete-confirmation`}
-                              ariaLabel={`Delete question: ${question.fieldTitle}`}
-                              dataCy={`cy_deleteQuestion-Section-${section.sectionTitle}-${question.fieldTitle}`}
+                              href={`/build-application/${applicationId}/${section.sectionId}`}
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/components/Sections.tsx:47]: ^

File: packages/admin/src/pages/build-application/[applicationId]/components/Sections.tsx [hunk 10]
```diff
@@ -175,17 +95,108 @@ const Sections = ({
...
-  applicationStatus: ApplicationFormSummary['applicationStatus'];
+function tableRows(
+  section: ApplicationFormSection,
+  applicationId: string,
+  applicationStatus: ApplicationFormSummary['applicationStatus'],
+  isSectionEligibilityOrEssential: boolean
+) {
+  let tableRows = section.questions
+    ? section.questions.map((question) => {
+        let link = `/build-application/${applicationId}/${section.sectionId}/${question.questionId}/preview`;
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/components/Sections.tsx:113]: ^

File: packages/admin/src/pages/build-application/[applicationId]/components/Sections.tsx [hunk 11]
```diff
@@ -175,17 +95,108 @@ const Sections = ({
...
+        // Eligibility is handled differently, so we explicitly direct to the eligibility-statement page here
+        if (section.sectionId === 'ELIGIBILITY') {
+          link = `/build-application/${applicationId}/${section.sectionId}/${question.questionId}/eligibility-statement`;
```
Comment (ConorFayleTCO) [inline packages/admin/src/pages/build-application/[applicationId]/components/Sections.tsx:116]: ^

**Review by ConorFayleTCO — APPROVED** [review]

File: packages/admin/src/pages/build-application/[applicationId]/[sectionId]/delete-confirmation.page.tsx (hunk 3 above)
Comment (dominicwest) [reply packages/admin/src/pages/build-application/[applicationId]/[sectionId]/delete-confirmation.page.tsx:64]: Nae routes file - this is the admin app

**Pull request merged**